integrationtests/src/test/java/io/kroxylicious/proxy/FilterIT.java

```diff
@@ -384,6 +384,24 @@ void requestFiltersCanRespondWithoutProxyingRespondsInCorrectOrder() throws Exce
         }
     }
 
+    @Test
+    void clientsCanSendMultipleMessagesImmediately() throws Exception {
+
+        try (var tester = mockKafkaKroxyliciousTester(s -> proxy(s).addToFilters(new FilterDefinitionBuilder("RejectingCreateTopic").build()));
+                var client = tester.simpleTestClient()) {
+            tester.addMockResponseForApiKey(new ResponsePayload(METADATA, METADATA.latestVersion(), new MetadataResponseData()));
+            tester.addMockResponseForApiKey(new ResponsePayload(API_VERSIONS, API_VERSIONS.latestVersion(), new ApiVersionsResponseData()));
+            Request requestA = new Request(METADATA, METADATA.latestVersion(), "clientA", new MetadataRequestData());
+            Request requestB = new Request(API_VERSIONS, API_VERSIONS.latestVersion(), "clientB", new ApiVersionsRequestData());
+            var futureA = client.get(requestA);
+            var futureB = client.get(requestB);
+            Response responseA = futureA.get(10, TimeUnit.SECONDS);
+            Response responseB = futureB.get(10, TimeUnit.SECONDS);
+            assertThat(responseA.payload()).isNotNull().extracting(ResponsePayload::apiKeys).isEqualTo(METADATA);
+            assertThat(responseB.payload()).isNotNull().extracting(ResponsePayload::apiKeys).isEqualTo(API_VERSIONS);
```

Comment:
I can see how this asserts two successful sends to the cluster, but I'm not seeing how this asserts that the requests didn't queue up or block. 

How did the old behaviour manifest?

Reply:
On main it causes a timeout in the `get` of the futures and the state machine is transitioned to `FAILED`.

Reply:
interesting, the state machine has several paths to this terminal FAILED state, but the inbound connection is only closed in one case, that's a failed connection to the upstream. Maybe we should make every FAILED transition close the inbound channel.

Reply:
> On main it causes a timeout in the `get` of the futures and the state machine is transitioned to `FAILED`.

Maybe add an assertion that the futures completed successfully rather than doing it implicitly by checking the values?

> Maybe we should make every FAILED transition close the inbound channel.


Intuitively it feels wrong that only one transition to failed kills the connection so I'd be in favour of updating it. Maybe @tombentley has some thoughts on the matter.